app/frontend/src/features/auth/Logout.tsx

```diff
@@ -1,11 +1,16 @@
 import { useEffect } from "react";
 import { logout } from "./authSlice";
-import { useAppDispatch } from "../../store";
+import { useAppDispatch, useTypedSelector } from "../../store";
+import { tokenLogout } from "./authActions";
 
 export default function Logout() {
   const dispatch = useAppDispatch();
+  const authToken = useTypedSelector((state) => state.auth.authToken);
 
   useEffect(() => {
+    if (authToken) {
+      tokenLogout(authToken);
```

Comment:
Functions created with `createAsyncThunk` should be dispatched instead of called directly.

I would rename this async thunk to just 'logout', then you can move the logic from the current logout action into the [authSlice extraReducers builder](https://github.com/Couchers-org/couchers/blob/65c2aff14dc1cbaffa4c7024114b3b9aa4d38db0/app/frontend/src/features/auth/authSlice.ts#L44) like the other asyncThunks. And then get rid of the dispatch two lines below this.

[Here's the docs](https://redux-toolkit.js.org/api/createAsyncThunk#type) which explains 'pending' 'fulfilled' and 'rejected'

Reply:
Thanks for the link! I definitely need to read the Redux docs more thoroughly. Still a lot I don't understand.